engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/DiagnosticException.java

```diff
@@ -0,0 +1,22 @@
+package org.enso.compiler.test.mock;
+
+import org.enso.compiler.core.ir.Diagnostic;
+
+public final class DiagnosticException extends RuntimeException {
```

Comment:
`public`? Is it good/necessary for having this exception public?

Reply:
It is used inside in a test: https://github.com/enso-org/enso/blob/9f46eea7de16489eac8704f9d32d512a372b7825/engine/runtime-compiler/src/test/java/org/enso/compiler/test/CompilerErrorTest.java#L31-L37

Could be theoretically replaced by using [CompilerError](https://github.com/enso-org/enso/blob/6abcbcb00c6aac11b506cfcf75d15683ef6f2fbc/engine/runtime-parser/src/main/java/org/enso/compiler/core/CompilerError.java) from `runtime-parser`, but the `DiagnosticException` gives more info.